include/libtorrent/bdecode.hpp

```diff
@@ -166,7 +166,7 @@ struct bdecode_token
 	enum type_t
 	{ none, dict, list, string, integer, end };
 
-	enum limits_t
+	enum limits_t : std::uint32_t
```

Comment:
since these are quantaties, meant for arithmetic, I don't think they should be unsigned. see this [blog post](http://blog.libtorrent.org/2016/05/unsigned-integers/).

Also, this is a bit of an abuse of enums. Ideally these constants should be ``constexpr`` constants.